### PR TITLE
Added very barebones VR support

### PR DESCRIPTION
**Pull request by kyle-emmerich**
This PR should give us rudimentary VR support. It is the bare minimum to avoid getting sick when using a VR headset, but some camera types should be avoided, such as Follow or Watch.

I also included a small typo fix. As far as I know, that's the only place that function shows up in the whole repository and it only exists in that small scope.

Here is a test place containing these changes: http://www.roblox.com/games/357912748/VR
Obviously, a VR headset is recommended for the best results.

### CODE REVIEW

**Comment by kyle-emmerich** [comment]
I also want to add that I've changed how the internal state tracking of the camera scripts work. If anyone sees a problem with this, let me know as soon as possible.

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript.rbxmx [hunk 1]
```diff
@@ -138,6 +138,8 @@ end
 local function Update()
 	if EnabledCamera then
 		EnabledCamera:Update()
+		EnabledCamera:ApplyCameraCFrame()
```
Comment (hockeychaos) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript.rbxmx]: Is there a reason we couldn't just include these as part of the Update() call?

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera.rbxmx [hunk 2]
```diff
@@ -72,6 +73,8 @@ end
 local function CreateCamera()
 	local this = {}
 
+	this.cframe = workspace.CurrentCamera and workspace.CurrentCamera.CoordinateFrame or CFrame.new()
```
Comment (hockeychaos) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera.rbxmx]: we should have kip look at this, he did most of the camera code

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera.rbxmx [hunk 3]
```diff
@@ -277,6 +280,18 @@ local function CreateCamera()
...
 
+	function this:ApplyCameraCFrame()
+		if workspace.CurrentCamera then
+			workspace.CurrentCamera.CoordinateFrame = this.cframe
+		end
+	end
+
+	function this:ApplyVROffset()
+		if UserInputService.IsVREnabled and workspace.CurrentCamera then
+			--workspace.CurrentCamera.CoordinateFrame = workspace.CurrentCamera.CoordinateFrame * UserInputService.UserHeadCFrame
```
Comment (hockeychaos) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera.rbxmx]: is this supposed to be commented out?

Comment (kyle-emmerich) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera.rbxmx]: Oh, oops. Let's just kill this PR, looks like I missed a couple things.

**Pull request closed without merging**